### PR TITLE
feat: (IA) dash brand header and <BoxContrast /> component

### PR DESCRIPTION
**Pull request by jaredrethman**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds brand header component to new Newspack Dashboard. Title text and background-color are derived dynamically from:
- Title = Site name
- Background color = Site Design > Header Background Color

This PR also introduces a new reusable component named `<BoxContrast />`:
* Allows passing in a hexadecimal color value which is evaluated to for luminescence.
* The relative luminescence will determine if white or black color should be used for best contrast.
* Hex color can be applied to `color` or `background-color` css props.
* Allows passing in text, or react component, to include within the box.

To aid in local ts testing I have added `npm run tsc` to create a typescript-compiler watch instance.

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets i.e. git `checkout feat/ia-dash-sections && npm run build`
2. Navigate to - /wp-admin/admin.php?page=newspack#/
3. Observe that Site Name and Background Color values have been applied to header.
4. Update Site Name and Background Color values and confirm they reflect in the header.

Below examples ("Newspack" is Site Name):

Screenshot 2024-03-28 at 11 52 01
Screenshot 2024-03-28 at 11 52 54
Screenshot 2024-03-28 at 11 54 07
Screenshot 2024-03-28 at 11 54 27

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/dashboard/components/brand-header.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+const BrandHeader = () => {
+	return (
+		<header
+			className="newspack-dashboard__brand-header"
+			style={ { backgroundColor: settings.headerBgColor } }
+		>
+			<BoxContrast
+				className="brand-header__inner"
+				content={ <h1>{ settings.siteName }</h1> }
```
Comment (dkoo) [inline assets/wizards/dashboard/components/brand-header.tsx]: Not a blocker at all, just wondering: why not just use `children` and nest the content inside `BoxContrast`?

File: package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
 		"start": "npm ci && newspack-scripts watch",
 		"watch": "newspack-scripts watch",
 		"test": "newspack-scripts test",
+		"tsc": "tsc --watch",
```
Comment (dkoo) [inline package.json:44]: Also not a blocker, just curious: how does this differ from the regular `watch` script? It looked to me like the TS is already getting compiled in that existing script.

**Review by dkoo — APPROVED** [review]
Looks great! No issues, just a couple of non-blocking questions.

File: assets/wizards/dashboard/components/brand-header.tsx (hunk 1 above)
Comment (jaredrethman) [reply assets/wizards/dashboard/components/brand-header.tsx]: That makes more sense - nice suggestion.

File: package.json (hunk 2 above)
Comment (jaredrethman) [reply package.json:44]: I found, that while the existing watch task does compile TS, it doesn't catch any issues until they hit CI. Having this essentially mimics the behavior of the [`newspack/check-typescript` job](https://github.com/Automattic/newspack-plugin/blob/trunk/.circleci/config.yml#L17) and displays any issues into the terminal.

File: assets/wizards/dashboard/components/brand-header.tsx (hunk 1 above)
Comment (jaredrethman) [reply assets/wizards/dashboard/components/brand-header.tsx]: Updated - https://github.com/Automattic/newspack-plugin/pull/3020/commits/b946d34bfa4e2dc6b5b6d84ac3f7ecaf6eb49030

**Comment by jaredrethman** [comment]
Thanks for the feedback @dkoo , have actioned your suggestions.

**Pull request merged**